### PR TITLE
Inference function of MMOCRAutoModel

### PR DESCRIPTION
**Pull request by Harry-zzh**
Implementation of MMOCR ([Kuang et al, (2021)](https://arxiv.org/pdf/2108.06543.pdf)) in automm for supporting ocr text detection and recognition simultaneously. This implementation supports single-image inference.
The model can be accessed through specifying the checkpoint's name. From [here](https://github.com/open-mmlab/mmocr#model-zoo) you can see the list of checkpoints can be used.

Usage:
```python
# text detection only
predictor = MultiModalPredictor(
        hyperparameters={
            "model.mmocr_text.det_ckpt_name": det_ckpt_name,
        },
        pipeline="ocr_text",
    )

pred = predictor.predict({"image": [mmocr_image_name]}) # ({"boundary_results":...})

# text recognition only (for images that only contain one word)
predictor = MultiModalPredictor(
        hyperparameters={
            "model.mmocr_text.recog_ckpt_name": recog_ckpt_name,
        },
        pipeline="ocr_text",
    )

pred = predictor.predict({"image": [mmocr_image_name]})  # ({'text':..., 'score': ...})

# text detection + text recognition (for images that contain multiple words)
predictor = MultiModalPredictor(
        hyperparameters={
            "model.mmocr_text.det_ckpt_name": det_ckpt_name,
            "model.mmocr_text.recog_ckpt_name": recog_ckpt_name,
        },
        pipeline="ocr_text",
    )

pred = predictor.predict({"image": [mmocr_image_name]})  # ({'text':...})
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/models/mmocr.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+        Returns
+        -------
+            A dictionary with bounding boxes.
+        """
+
+        data = batch[self.image_key]
+        # single image
+        if isinstance(data["img_metas"], List):
+            data["img_metas"] = [img_metas.data[0] for img_metas in data["img_metas"]]
```
Comment (FANGAreNotGnu) [inline multimodal/src/autogluon/multimodal/models/mmocr.py:133]: Data unpacks and to(device) shall be moved to collate function (maybe in next pr). For details see #2244.

File: multimodal/tests/unittests/test_mmocr_text_detection.py [hunk 2]
```diff
@@ -42,10 +42,10 @@ def test_mmocr_text_detection_inference(checkpoint_name):
     MMOCR_res = ocr.readtext(mmocr_image_name, output=None)
 
     # compare the outputs of original model's output and our model
-    assert len(pred) == len(MMOCR_res[0]["boundary_result"])  # num of text lines
+    assert len(pred[0]) == len(MMOCR_res[0]["boundary_result"])  # num of text lines
```
Comment (FANGAreNotGnu) [inline multimodal/tests/unittests/test_mmocr_text_detection.py:45]: If the first dimension of pred result is not used, we may consider remove it in extract_from_output

**Review by FANGAreNotGnu — APPROVED** [review]
LGTM with a minor comment, and a TODO for following PR. Thanks!

File: multimodal/tests/unittests/test_mmocr_text_detection.py (hunk 2 above)
Comment (Harry-zzh) [reply multimodal/tests/unittests/test_mmocr_text_detection.py]: Done.

**Pull request closed without merging**